File: backend/api-service/src/main/kotlin/com/sideproject/api/archive/entity/Archive.kt
package com.sideproject.api.archive.entity

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(
    name = "archive",
    indexes = [
        Index(name = "idx_archive_user", columnList = "user_id"),
        Index(name = "idx_archive_folder", columnList = "folder_id"),
        Index(name = "idx_archive_user_folder", columnList = "user_id,folder_id")
    ]
)
class Archive(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @Column(name = "user_id", nullable = false)
    val userId: Long,

    @Column(name = "folder_id", nullable = false)
    var folderId: Long,

    @Column(nullable = false, length = 1000)
    val url: String,

    @Column(name = "use_yn", nullable = false, length = 1)
    var useYn: String = "Y", // 기본값 'Y' 설정

    @Column(length = 255)
    var title: String? = null,

    @Column(name = "thumbnail_url", columnDefinition = "TEXT")
    val thumbnailUrl: String? = null,

    @Column(name = "ai_summary", columnDefinition = "TEXT") // 혹은 아예 TEXT 타입으로 지정
    var aiSummary: String? = null,

    @Column(name = "created_at", nullable = false, updatable = false)
    val createdAt: LocalDateTime = LocalDateTime.now()


) {

    @OneToMany(
        mappedBy = "archive",
        fetch = FetchType.LAZY,
    )
    val archiveKeywords: MutableList<ArchiveKeyword> = mutableListOf()

    fun clearKeywords() {
        this.archiveKeywords.clear()
    }

    // 수정 가능 필드 업데이트 로직
    fun update(title: String, aiSummary: String?, folderId: Long) {
        this.title = title
        this.aiSummary = aiSummary
        this.folderId = folderId
    }

    // soft delete 로직
    fun delete() {
        this.useYn = "N"
    }

    // Archive.kt 내부
    fun addArchiveKeyword(archiveKeyword: ArchiveKeyword) {
        this.archiveKeywords.add(archiveKeyword)
    }
}

File: frontend/src/types/archive.ts
export interface ArchiveResponse {
    id: number;
    userId: number;
    folderId: number;
    url: string;
    title: string | null;         // null 허용 (서버 title: String?)
    thumbnailUrl: string | null;  // null 허용
    aiSummary: string | null;     // null 허용
    keywords: string[];           // 가공된 문자열 리스트
    createdAt: string;            // LocalDateTime은 JSON 직렬화 시 ISO 8601 문자열로 변환됨
}

export interface ArchiveMetadata {
    title?: string;
    thumbnailUrl?: string;
}

export interface ArchiveAiAnalyze {
    summary?: string;
    keywords?: string[];
}

export interface ArchiveCreateRequest {
    url: string;
    title: string;
    thumbnailUrl: string | null;
    aiSummary: string | null;
    folderId: number;
    keywords: string[] | null;
}

export interface ArchiveUpdateRequest {
    title: string;
    aiSummary: string | null;
    folderId: number;
    keywords: string[] | null;
}